clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -4276,13 +4276,18 @@ EmitConditionalOperatorLValue(const AbstractConditionalOperator *expr) {
     return EmitUnsupportedLValue(expr, "conditional operator");
   rhsBlock = Builder.GetInsertBlock();
 
-  EmitBlock(contBlock);
-
   if (lhs && rhs) {
-    llvm::PHINode *phi = Builder.CreatePHI(lhs->getPointer()->getType(),
-                                           2, "cond-lvalue");
-    phi->addIncoming(lhs->getPointer(), lhsBlock);
-    phi->addIncoming(rhs->getPointer(), rhsBlock);
+    llvm::Value *lhsPtr = lhs->getPointer();
+    llvm::Value *rhsPtr = rhs->getPointer();
+    if (rhsPtr->getType() != lhsPtr->getType())
```

Comment:
JIC, is it guaranteed that `rhsPtr`/`lhsPtr` are non-`nullptr`?

Reply:
I think it is. In any case I am not changing behavior with regards to this part. So if it was or was not guaranteed earlier the same behavior remains now.